src/Defer/DeferrableDirective.php

```diff
@@ -55,42 +55,41 @@ public function handleField(FieldValue $value, Closure $next): FieldValue
         $fieldType = $value->getField()->type;
 
         $value->setResolver(
-            function ($root, $args, GraphQLContext $context, ResolveInfo $info) use ($resolver, $fieldType) {
-                $path = implode('.', $info->path);
-                $extension = $this->getDeferExtension();
-                $wrappedResolver = function () use ($resolver, $root, $args, $context, $info) {
-                    return $resolver($root, $args, $context, $info);
+            function ($root, array $args, GraphQLContext $context, ResolveInfo $resolveInfo) use ($resolver, $fieldType) {
+                $wrappedResolver = function () use ($resolver, $root, $args, $context, $resolveInfo) {
+                    return $resolver($root, $args, $context, $resolveInfo);
                 };
+                $path = implode('.', $resolveInfo->path);
 
-                if ($this->shouldDefer($fieldType, $info)) {
-                    return $extension->defer($wrappedResolver, $path);
+                if ($this->shouldDefer($fieldType, $resolveInfo)) {
+                    return $this->defer->defer($wrappedResolver, $path);
                 }
 
-                return $extension->isStreaming()
-                    ? $extension->findOrResolve($wrappedResolver, $path)
-                    : $resolver($root, $args, $context, $info);
+                return $this->defer->isStreaming()
+                    ? $this->defer->findOrResolve($wrappedResolver, $path)
+                    : $resolver($root, $args, $context, $resolveInfo);
             }
         );
 
         return $next($value);
     }
 
     /**
-     * Determine of field should be deferred.
+     * Determine if field should be deferred.
      *
      * @param  \GraphQL\Language\AST\TypeNode  $fieldType
-     * @param  \GraphQL\Type\Definition\ResolveInfo  $info
+     * @param  \GraphQL\Type\Definition\ResolveInfo  $resolveInfo
      * @return bool
      *
      * @throws \Nuwave\Lighthouse\Exceptions\ParseClientException
      */
-    protected function shouldDefer(TypeNode $fieldType, ResolveInfo $info): bool
+    protected function shouldDefer(TypeNode $fieldType, ResolveInfo $resolveInfo): bool
     {
-        if (strtolower($info->operation->operation) === 'mutation') {
+        if (strtolower($resolveInfo->operation->operation) === 'mutation') {
```

Comment:
Maybe change this to also use `Str::lower()` instead as we use `Str` lots of other places.
The only reason for not to do it, is that this is a base php method.

Reply:
Not sure if this is better, maybe you can open up an issue about this?

Reply:
https://github.com/nuwave/lighthouse/issues/653